### PR TITLE
Stackpulse integration

### LINKED ISSUE #9870: Support StackPulse
##### Feature idea summary
Create alerts in StackPulse

More details will come after to study API.

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fixes #9870

##### Component Name
Health
##### Test Plan

1 - Ask me about the Integration hook.

hook

2 - Configure your `health_alarm_notify.conf` with the following variables:

```
SEND_STACKPULSE="YES"
STACKPULSE_WEBHOOK="https://hooks.stackpulse.io/v1/webhooks/UUID_HERE"
```
 
3 - Go to directory `/usr/libexec/netdata/plugins.d/` and run:

```
./alarm-notify.sh test
```

4 - Check on Stackpulse dashboard Journal for messages like this:

alarm

##### Additional Information
You can also configure a Playbook to organize the alarms:

playbook

playbook
image

### CODE REVIEW

File: health/notifications/alarm-notify.sh.in [hunk 1]
```diff
@@ -2016,6 +2027,36 @@ EOF
...
+    "Description" : "${status_message}, ${info}"
+  }
+EOF
+)
+
+  httpcode=$(docurl -X POST -H "Content-Type: application/json" -d "${payload}" ${STACKPULSE_WEBHOOK})
+  if [ "${httpcode}" = "200" ]; then
+    info "sent stackpulse notification for: ${host} ${chart}.${name} is ${status}"
+  else
+    error "failed to send stackpulse notification for: ${host} ${chart}.${name} is ${status}, with HTTP error code ${httpcode}."
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:2055]: I agree with  you that your suggestion is better than the initial sentence, but it is the current pattern:

```
grep "HTTP error code " health/notifications/alarm-notify.sh.in 
                                error "failed to send pushover notification for: ${host} ${chart}.${name} is ${status} to '${user}' with HTTP error code ${httpcode}."
                                error "failed to send pushbullet notification for: ${host} ${chart}.${name} is ${status} to '${user}' with HTTP error code ${httpcode}."
                        error "failed to send kafka data for: ${host} ${chart}.${name} is ${status} and ip '${KAFKA_SENDER_IP}' with HTTP error code ${httpcode}."
                                error "failed to send pagerduty notification for: ${host} ${chart}.${name} is ${status}, with HTTP error code ${httpcode}."
                                error "failed to send Twilio SMS for: ${host} ${chart}.${name} is ${status} to '${user}' with HTTP error code ${httpcode}."
                                error "failed to send HipChat notification for: ${host} ${chart}.${name} is ${status} to '${room}' with HTTP error code ${httpcode}."
                                error "failed to send Messagebird SMS for: ${host} ${chart}.${name} is ${status} to '${user}' with HTTP error code ${httpcode}."
                                error "failed to send Kavenegar SMS for: ${host} ${chart}.${name} is ${status} to '${user}' with HTTP error code ${httpcode}."
                                error "failed to send telegram notification for: ${host} ${chart}.${name} is ${status} to '${chatid}' with HTTP error code ${httpcode}."
                        error "failed to send Microsoft team notification for: ${host} ${chart}.${name} is ${status} to '${webhook}', with HTTP error code ${httpcode}."
                        error "failed to send slack notification for: ${host} ${chart}.${name} is ${status} ${chstr}, with HTTP error code ${httpcode}."
                        error "failed to send rocketchat notification for: ${host} ${chart}.${name} is ${status} to '${channel}', with HTTP error code ${httpcode}."
                        error "failed to send alerta notification for: ${host} ${chart}.${name} is ${status} to '${channel}', with HTTP error code ${httpcode}."
                        error "failed to send flock notification for: ${host} ${chart}.${name} is ${status} to '${channel}', with HTTP error code ${httpcode}."
                        error "failed to send discord notification for: ${host} ${chart}.${name} is ${status} to '${channel}', with HTTP error code ${httpcode}."
                                error "failed to send fleep data for: ${host} ${chart}.${name} is ${status} and user '${FLEEP_SENDER}' with HTTP error code ${httpcode}."
                        error "failed to send Matrix notification for: ${host} ${chart}.${name} is ${status} to '${room}', with HTTP error code ${httpcode}."
                                error "failed to send hangouts notification for: ${host} ${chart}.${name} is ${status} to '${room}', with HTTP error code ${httpcode}."
```

I think we need a PR to bring the same pattern for the whole script.

[the remark this answers is not in the document]
Comment (ilyam8) [reply health/notifications/alarm-notify.sh.in:2055]: I think my suggestion is minor and could be ignored but it took my attention reading the code. I agree with a separate PR for it.

[the remark this answers is not in the document]
Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:2055]: Of course, I will change this PR as you suggested, and I will make another to create an unique pattern.

File: health/notifications/alarm-notify.sh.in [hunk 2]
```diff
@@ -2016,6 +2027,50 @@ EOF
...
+    "ID" : ${alarm_id}
+  }
+EOF
+)
+
+  httpcode=$(docurl -X POST -H "Content-Type: application/json" -d "${payload}" ${STACKPULSE_WEBHOOK})
+  if [ "${httpcode}" = "200" ]; then
+    info "sent stackpulse notification for: ${host} ${chart}.${name} is ${status}"
+  else
+    error "failed to send stackpulse notification for: ${host} ${chart}.${name} is ${status}, with HTTP response status code ${httpcode}."
```
Comment (ilyam8) [inline health/notifications/alarm-notify.sh.in:2069]: returni think `else` branch should `return 1`

File: health/notifications/alarm-notify.sh.in [hunk 3]
```diff
@@ -2561,7 +2621,8 @@ for state in "${SENT_EMAIL}" \
 	"${SENT_SYSLOG}" \
 	"${SENT_SMS}" \
 	"${SENT_MSTEAM}" \
-    "${SENT_DYNATRACE}"; do	
+  "${SENT_DYNATRACE}" \
+  "${SENT_STACKPULSE}"; do
```
Comment (ilyam8) [inline health/notifications/alarm-notify.sh.in:2625]: minor, but align please 😄

File: health/notifications/alarm-notify.sh.in (hunk 2 above)
Comment (Ferroin) [reply health/notifications/alarm-notify.sh.in:2069]: Yes, it should return 1, otherwise we'll always think we sent notifications successfully.

Comment (thiagoftsm) [reply health/notifications/alarm-notify.sh.in:2069]: Thank you guys! I inserted the return.

**Review by ilyam8 — DISMISSED** [review]

**Comment by thiagoftsm** [comment]
@ilyam8 the alignment was a Clion "gift", I used vi to fix it.

**Review by Ferroin — APPROVED** [review]
Approved based on code review and information about testing that was discussed on Slack.

**Review by ilyam8 — APPROVED** [review]

**Comment by ilyam8** [comment]
the target branch is master atm

**Comment by prologic** [comment]
> the target branch is master atm

Fixed.

**Comment by ilyam8** [comment]
@thiagoftsm ping

**Comment by thiagoftsm** [comment]
@prologic this PR was waiting for the code freeze to be finished, it was already ready to be merged. Why was this moved to develop and after this closed?

**Comment by ilyam8** [comment]
@thiagoftsm 

i think because of https://github.com/netdata/netdata/pull/9886#issuecomment-693335068

i think we should have it merged into the develop, it had 2 approval reviews 5 days ago

**Pull request closed without merging**